cloud/client.go

```diff
@@ -1111,6 +1165,385 @@ func (c *client) DeleteSatellite(ctx context.Context, name, orgID string) error
+	if err != nil {
+		return nil, err
+	}
+
+	if status != http.StatusCreated {
+		return nil, errors.Errorf("failed to create project: %s", body)
+	}
+
+	res := &secretsapi.CreateProjectResponse{}
+
+	err = jsonpb.UnmarshalString(body, res)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Project{
+		Name:       res.Project.Name,
+		OrgName:    res.Project.OrgName,
+		CreatedAt:  res.Project.CreatedAt.AsTime(),
+		ModifiedAt: res.Project.ModifiedAt.AsTime(),
+	}, nil
+}
+
+// ListProjects returns all projects in the organization that are visible to the
+// logged-in user.
+func (c *client) ListProjects(ctx context.Context, orgName string) ([]*Project, error) {
+	u := fmt.Sprintf("/api/v0/projects/%s", orgName)
+
+	status, body, err := c.doCall(ctx, http.MethodGet, u, withAuth())
+	if err != nil {
+		return nil, err
+	}
+
+	if status != http.StatusOK {
+		return nil, errors.Errorf("failed to list projects: %s", body)
+	}
+
+	res := &secretsapi.ListProjectsResponse{}
+
+	err = jsonpb.UnmarshalString(body, res)
+	if err != nil {
+		return nil, err
+	}
+
+	var projects []*Project
+
+	for _, pro := range res.Projects {
+		projects = append(projects, &Project{
+			Name:       pro.Name,
+			OrgName:    pro.OrgName,
+			CreatedAt:  pro.CreatedAt.AsTime(),
+			ModifiedAt: pro.ModifiedAt.AsTime(),
+		})
+	}
+
+	return projects, nil
+}
+
+// GetProject loads a single project from the projects endpoint.
+func (c *client) GetProject(ctx context.Context, orgName, name string) (*Project, error) {
+	u := fmt.Sprintf("/api/v0/projects/%s/%s", orgName, name)
+
+	status, body, err := c.doCall(ctx, http.MethodPost, u, withAuth())
+	if err != nil {
+		return nil, err
+	}
+
+	if status != http.StatusOK {
+		return nil, errors.Errorf("failed to get project: %s", body)
+	}
+
+	res := &secretsapi.GetProjectResponse{}
+
+	err = jsonpb.UnmarshalString(body, res)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Project{
+		Name:       res.Project.Name,
+		OrgName:    res.Project.OrgName,
+		CreatedAt:  res.Project.CreatedAt.AsTime(),
+		ModifiedAt: res.Project.ModifiedAt.AsTime(),
+	}, nil
+}
+
+// DeleteProject deletes a given project by name.
+func (c *client) DeleteProject(ctx context.Context, orgName, name string) error {
+	u := fmt.Sprintf("/api/v0/projects/%s/%s", orgName, name)
+
+	status, body, err := c.doCall(ctx, http.MethodDelete, u, withAuth())
+	if err != nil {
+		return err
+	}
+
+	if status != http.StatusOK {
+		return errors.Errorf("failed to delete project: %s", body)
+	}
+
+	return nil
+}
+
+// AddProjectMember adds a new member to the project by email or user ID.
+func (c *client) AddProjectMember(ctx context.Context, orgName, name, idOrEmail, permission string) error {
+	u := fmt.Sprintf("/api/v0/projects/%s/%s", orgName, name)
+
+	req := &secretsapi.AddProjectMemberRequest{
+		Permission: permission,
+	}
+
+	if userID, err := uuid.Parse(idOrEmail); err == nil && userID != uuid.Nil {
```

Comment:
Is it possible to move this `idOrEmail` logic into a separate function, and then extend all the project functions (e.g. `UpdateProjectMember`, `ListProjectMembers`, etc), to all accept an email or ID?

Reply:
I think that's only used for that one endpoint/method. The other ones assume a userID, which is returned from the member list endpoint.